src/quicktype-core/language/Crystal.ts

```diff
@@ -0,0 +1,429 @@
+import { TargetLanguage } from "../TargetLanguage";
+import { ConvenienceRenderer, ForbiddenWordsInfo } from "../ConvenienceRenderer";
+import {
+    legalizeCharacters,
+    splitIntoWords,
+    isLetterOrUnderscoreOrDigit,
+    combineWords,
+    allLowerWordStyle,
+    firstUpperWordStyle,
+    intToHex,
+    utf32ConcatMap,
+    escapeNonPrintableMapper,
+    isPrintable,
+    isAscii,
+    isLetterOrUnderscore
+} from "../support/Strings";
+import { Name, Namer, funPrefixNamer } from "../Naming";
+import { UnionType, Type, ClassType, EnumType } from "../Type";
+import { matchType, nullableFromUnion, removeNullFromUnion } from "../TypeUtils";
+import { Sourcelike, maybeAnnotated } from "../Source";
+import { anyTypeIssueAnnotation, nullTypeIssueAnnotation } from "../Annotation";
+import { Option } from "../RendererOptions";
+import { RenderContext } from "../Renderer";
+
+export class CrystalTargetLanguage extends TargetLanguage {
+    protected makeRenderer(renderContext: RenderContext): CrystalRenderer {
+        return new CrystalRenderer(this, renderContext);
+    }
+
+    constructor() {
+        super("crystal", ["crystal", "cr", "crystallang"], "cr");
+    }
+
+    protected get defaultIndentation(): string {
+        return "  ";
```

Comment:
Only two spaces?  Is that the convention in Crystal?

Reply:
Yes https://crystal-lang.org/docs/conventions/coding_style.html